Gemfile

```
@@ -105,6 +105,7 @@ group :development, :test do
   gem 'bullet', '~> 7.0'
   gem 'capybara-webmock', git: 'https://github.com/hashrocket/capybara-webmock.git', ref: 'd3f3b7c'
   gem 'erb_lint', '~> 0.7.0', require: false
+  gem 'faker'
```

let's add an explicit require in the mailer preview too? or maybe a comment here noting why it's here?

```suggestion
  gem 'faker', require: false
```

---

the comment is to make sure it doesn't get removed in the future

the `require: false` is because we don't strictly need that in memory for our "normal" production flow so this just helps keep things small and explicit

---

Unless I'm misunderstanding something, moving from `:test` to the `:development, :test` group shouldn't impact production?

---

I think the idea is to avoid auto-requiring the gem via Bundler when running the application locally, which wouldn't have happened previously but will happen if it's moved into the `development` group.

https://github.com/18F/identity-idp/blob/c1cda40e189ec20c0d8dec77f4af5dbbf8a325dd/config/application.rb#L23

```
$ rails c
[1] pry(main)> Rails.groups
=> [:default, "development"]
```

---

Yeah, that aspect makes sense. I wasn't sure I understood where the production aspect fit in.

---

Hey, I'm not sure the juice is worth the squeeze on this change.  After adding `require: false`, I realized that it caused `NameError:  uninitialized constant Faker` errors when running specs that uses faker, e.g. `bundle exec rspec spec/forms/idv/address_form_spec.rb`.  

I tried to add `require 'faker'` to the `rails_helper` below [line 16](https://github.com/18F/identity-idp/blob/main/spec/rails_helper.rb#L16), but that didn't resolve the error.  Then I tried adding `require 'faker'` to the [`spec_helper`](https://github.com/18F/identity-idp/blob/b1be266d5c98a9438bbd568e37aacac6caf64988/spec/spec_helper.rb#L11), but that also didn't resolve the error.

Is there a way to get faker required everywhere it's used in the codebase specs without having to add `require 'faker'` to the top of each individual file?

---

I am surprised that `spec_helper` wasn't enough, let me check out this branch and noodle on it. But for the sake of expediency and fixing the build, I wouldn't hold up this PR on `require: false`

---

Thanks, @zachmargolis!  Yeah, I was also really surprised that including it in `spec_helper` didn't help....  🤔

---

I tried another approach entirely in https://github.com/18F/identity-idp/pull/11791, which is just to remove usage of `Faker` in the mailer preview